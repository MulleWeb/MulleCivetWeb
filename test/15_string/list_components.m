#import <MulleCivetWeb/MulleCivetWeb.h>


void test( NSString *s, NSString *component, NSString *separator)
{
   NSRange    range;
   NSString   *added;
   NSString   *removed;

   if( s)
      printf( "s=%s ", [s UTF8String]);
   else
      printf( "s=nil ");
   if( component)
      printf( "component=%s ", [component UTF8String]);
   else
      printf( "component=nil ");
   if( separator)
      printf( "separator=%s ", [separator UTF8String]);
   else
      printf( "separator=nil ");


   range = [s mulleRangeOfListComponent:component
                              separator:separator];
   if( range.length)
      printf( "range = { %lu, %lu } ", (unsigned long) range.location,
                                      (unsigned long) range.length);
   else
      printf( "range = { NSNotFound, 0 } ");

   added = [s mulleStringByAddingListComponent:component
                                     separator:separator];
   if( added)
      printf( "added=%s ", [added UTF8String]);
   else
      printf( "added=nil ");

   removed = [s mulleStringByRemovingListComponent:component
                                         separator:separator];

   if( removed)
      printf( "removed=%s ", [removed UTF8String]);
   else
      printf( "removed=nil ");
   printf( "\n");
}

void  test_separator_empty( NSString *separator)
{
   test( @"", nil, separator);
   test( @"", @"", separator);

   printf( "\n");
}


void  test_separator_comma( NSString *separator)
{
   test( @"", @"a", separator);
   test( @"a", @"", separator);
   test( @"a", @"a", separator);
   test( @"a", @"b", separator);
   test( @"a,b", @"c", separator);

   test( @"a,b,c", @"", separator);
   test( @"a,b,c", @"a", separator);
   test( @"a,b,c", @"b", separator);
   test( @"a,b,c", @"c", separator);
   test( @"a,b,c", @"d", separator);

   printf( "\n");
}


void  test_separator_minusminus( NSString *separator)
{
   test( @"a--b", @"c", separator);

   test( @"a--b--c", @"", separator);
   test( @"a--b--c", @"a", separator);
   test( @"a--b--c", @"b", separator);
   test( @"a--b--c", @"c", separator);
   test( @"a--b--c", @"d", separator);

   printf( "\n");
}


int   main( int argc, char *argv[])
{
   test_separator_empty( nil);
   test_separator_empty( @"");
   test_separator_empty( @",");

   test_separator_comma( @",");
   test_separator_minusminus( @"--");

   return( 0);
}
