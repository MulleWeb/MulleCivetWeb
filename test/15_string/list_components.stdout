s= component=nil separator=nil range = { NSNotFound, 0 } added= removed= 
s= component= separator=nil range = { NSNotFound, 0 } added= removed= 

s= component=nil separator= range = { NSNotFound, 0 } added= removed= 
s= component= separator= range = { NSNotFound, 0 } added= removed= 

s= component=nil separator=, range = { NSNotFound, 0 } added=nil removed= 
s= component= separator=, range = { NSNotFound, 0 } added= removed= 

s= component=a separator=, range = { NSNotFound, 0 } added=a removed= 
s=a component= separator=, range = { NSNotFound, 0 } added=a, removed=a 
s=a component=a separator=, range = { 0, 1 } added=a removed= 
s=a component=b separator=, range = { NSNotFound, 0 } added=a,b removed=a 
s=a,b component=c separator=, range = { NSNotFound, 0 } added=a,b,c removed=a,b 
s=a,b,c component= separator=, range = { NSNotFound, 0 } added=a,b,c, removed=a,b,c 
s=a,b,c component=a separator=, range = { 0, 1 } added=a,b,c removed=b,c 
s=a,b,c component=b separator=, range = { 2, 1 } added=a,b,c removed=a,c 
s=a,b,c component=c separator=, range = { 4, 1 } added=a,b,c removed=a,b 
s=a,b,c component=d separator=, range = { NSNotFound, 0 } added=a,b,c,d removed=a,b,c 

s=a--b component=c separator=-- range = { NSNotFound, 0 } added=a--b--c removed=a--b 
s=a--b--c component= separator=-- range = { NSNotFound, 0 } added=a--b--c-- removed=a--b--c 
s=a--b--c component=a separator=-- range = { 0, 1 } added=a--b--c removed=b--c 
s=a--b--c component=b separator=-- range = { 3, 1 } added=a--b--c removed=a--c 
s=a--b--c component=c separator=-- range = { 6, 1 } added=a--b--c removed=a--b 
s=a--b--c component=d separator=-- range = { NSNotFound, 0 } added=a--b--c--d removed=a--b--c 

